src/test/java/uk/nhs/hee/tis/trainee/credentials/api/PlacementResourceTest.java

```diff
@@ -0,0 +1,160 @@
+  private static final String TRAINEE_ID_1 = "47165";
+  private static final String TIS_ID_1 = "2bd3f009-9c7a-4352-bcc6-9dff8eac07d6";
+  private static final String CREDENTIAL_TYPE_1 = "TRAINING_PROGRAMME";
+  private static final String ISSUED_AT_1 = "2017-02-03T10:37:30Z";
+  private static final String REVOKED_AT_1 = "2018-02-03T10:37:30Z";
+  private static final String CREDENTAIL_ID_2 = "e2a26a50-2c51-4327-8692-1cb05g32af22";
+  private static final String TRAINEE_ID_2 = "47165";
+  private static final String TIS_ID_2 = "2bd3f009-9c7a-4352-bcc6-9dff8eac07u9";
+  private static final String CREDENTIAL_TYPE_2 = "TRAINING_PROGRAMME";
+  private static final String ISSUED_AT_2 = "2011-02-03T10:37:30Z";
+  private static final Instant REVOKED_AT_2 = null;
+
+  /**
+   * Set up mocks before each test.
+   */
+  @BeforeEach
+  void setup() {
+    PlacementResource placementResource = new PlacementResource(service,
+        new CredentialMetadataMapperImpl());
+    this.mockMvc = MockMvcBuilders.standaloneSetup(placementResource)
+        .setMessageConverters(jacksonMessageConverter)
+        .build();
+
+    cred1.setCredentialId(CREDENTAIL_ID_1);
+    cred1.setTraineeId(TRAINEE_ID_1);
+    cred1.setTisId(TIS_ID_1);
+    cred1.setCredentialType(CREDENTIAL_TYPE_1);
+    cred1.setIssuedAt(Instant.parse(ISSUED_AT_1));
+    cred1.setRevokedAt(Instant.parse(REVOKED_AT_1));
+
+    cred2.setCredentialId(CREDENTAIL_ID_2);
+    cred2.setTraineeId(TRAINEE_ID_2);
+    cred2.setTisId(TIS_ID_2);
+    cred2.setCredentialType(CREDENTIAL_TYPE_2);
+    cred2.setIssuedAt(Instant.parse(ISSUED_AT_2));
+    cred2.setRevokedAt(REVOKED_AT_2);
+  }
+
+  @Test
+  void testSuccessfullyCallTheService() throws Exception {
+
+    String tisId = "47165";
+    String token = TestJwtUtil.generateTokenForTisId(tisId);
+
+    when(service.getTheLatestCredential(CredentialType.TRAINING_PLACEMENT, tisId))
+        .thenReturn(List.of(cred1));
+
+    mockMvc.perform(get("/api/placement")
+            .header(HttpHeaders.AUTHORIZATION, token)
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk())
+        .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+        .andExpect(jsonPath("$.[*].credentialId").value(CREDENTAIL_ID_1))
+        .andExpect(jsonPath("$.[*].traineeId").value(TRAINEE_ID_1))
+        .andExpect(jsonPath("$.[*].tisId").value(TIS_ID_1))
+        .andExpect(jsonPath("$.[*].credentialType").value(CREDENTIAL_TYPE_1))
+        .andExpect(jsonPath("$.[*].issuedAt").value(ISSUED_AT_1))
+        .andExpect(jsonPath("$.[*].revokedAt").value(REVOKED_AT_1));
+  }
+
+  @Test
+  void testReturnListOfResultsWithCorrectValues() throws Exception {
+    String tisId = "47165";
+    String token = TestJwtUtil.generateTokenForTisId(tisId);
+
+    when(service.getTheLatestCredential(CredentialType.TRAINING_PLACEMENT, tisId))
+        .thenReturn(List.of(cred1, cred2));
```

Comment:
Maybe this is a bit nitpicky, but cred2 has type TRAINING_PROGRAMME, so it would never be returned according to the given filter - I'd consider simply changing its type to make the logic of the mocked call consistent with what would be returned in reality.

Reply:
In fact the both should be TRAINING_PLACEMENT